.github/workflows/build.yml

```diff
@@ -11,12 +11,146 @@ on:
   workflow_dispatch:
 
 jobs:
+
+  setup:
+    runs-on: ubuntu-latest
+    outputs: 
+      PACKAGE_VERSION: ${{ steps.builddata.outputs.PACKAGE_VERSION }}
+      MINOR_PACKAGE_VERSION: ${{ steps.builddata.outputs.MINOR_PACKAGE_VERSION }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+          cache: 'gradle'
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+
+      - name: BuildData
+        id: builddata
+        run: |
+          set -xev
+          env | sort
+          
+          # handle full version number
+          VERSION=$(./gradlew -q printVersionName | head -n 1 | cut -d'-' -f1)  
+          VERSION=${VERSION// }
+          echo Current version in code is :${VERSION}:
+          echo "PACKAGE_VERSION=${VERSION}" >> $GITHUB_OUTPUT
+          # handle minor version
+          MINOR_VERSION=$(echo $VERSION | sed 's/^\([0-9]\{1,\}\.[0-9]\{1,\}\)\..*/\1/')
+          MINOR_VERSION=${MINOR_VERSION// }
+          echo Current minor version in code is :${MINOR_VERSION}:
+          echo "MINOR_PACKAGE_VERSION=${MINOR_VERSION}" > $GITHUB_OUTPUT
+      
+
+
   build:
     runs-on: ubuntu-latest
+    needs: [setup]
     steps:
     - uses: actions/checkout@v3
     - uses: actions/setup-java@v3
       with:
         distribution: 'temurin'
         java-version: '11'
         cache: 'gradle'
+    - name: Validate Gradle wrapper
+      uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+    - name: Build with Gradle
+      uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
+      with:
+        arguments: build        
+    - name: Create artifacts
+      env:
+        SIGNING_ID: ${{ secrets.SIGNING_ID }}
+        SIGNING_PASSWORD: ${{ secrets.SIGNING_PASSWORD }}
+        KEYRING_FILE: ${{ secrets.KEYRING_FILE }}
+      run: |
+        set -xev
+
+        echo ${{ needs.setup.steps.builddata.outputs.PACKAGE_VERSION }}
+        echo ${{ needs.setup.steps.builddata.outputs.MINOR_PACKAGE_VERSION }}
+
+        # note need to check this works... https://discuss.circleci.com/t/gpg-keys-as-environment-variables/28641
+        echo $KEYRING_FILE > keyringfile.gpg
+
+        ./gradlew -Psigning.keyId=${SIGNING_ID} -Psigning.password=${SIGNING_PASSWORD} -Psigning.secretKeyRingFile=keyringfile.gpg publishShimJarPublicationToMavenLocal
+        
+        tar -zcvf localmaven.tgz ${HOME}/.m2/repository/org/hyperledger
+        docker image save hyperledger/fabric-javaenv | gzip > fabric-javaenv.tar.gz
+    - name: Upload artifacts (1)
```

Comment:
It would be nice to include what's being uploaded in the name instead of `(1)` and `(2)` below